lib/buffer_update_component.coffee

```diff
@@ -5,11 +6,14 @@ module.exports =
 class BufferUpdateComponent
   constructor: (@editor, @analysisAPI) ->
     @events = new CompositeDisposable()
-    @events.add @editor.onDidStopChanging =>
+    updateFile = =>
       descriptors = @editor.getRootScopeDescriptor()
       # We only support pure Dart files for now
       return unless Utils.isCompatible(editor)
       @analysisAPI.updateFile @editor.getPath(), @editor.getText()
 
+    @events.add @editor.onDidStopChanging _.debounce(updateFile, 250,
+      maxWait: 1000)
```

Comment:
What about 5sec or 7sec for the max wait time? That way if they're still typing we won't analyze, unless it's been pretty long since the last analysis.